great_expectations/data_context/data_context.py

```diff
@@ -670,6 +672,78 @@ def update_return_obj(self, data_asset, return_obj):
         else:
             return return_obj
 
+    def profile_datasource(self, datasource_name, profiler_name="PseudoPandasProfiling", max_data_assets=10):
+        # logger.info("Profiling %s with %s" % (datasource_name, profiler_name))
+        print("Profiling %s with %s" % (datasource_name, profiler_name))
+        datasource = self.get_datasource(datasource_name)
+        data_asset_names = datasource.get_available_data_asset_names()
+
+        #!!! Abe 2019/06/11: This seems brittle. I don't understand why this object is packaged this way.
+        #!!! Note: need to review this to make sure the names are properly qualified.
+        data_asset_name_list = list(data_asset_names[0]["available_data_asset_names"])
+        total_data_assets = len(data_asset_name_list)
+        # logger.info("Found %d named data assets" % (total_data_assets))
+        print("Found %d named data assets" % (total_data_assets))
+        
+        if max_data_assets == None or max_data_assets >= len(data_asset_name_list):
+            # logger.info("Profiling all %d." % (len(data_asset_name_list)))
+            print("Profiling all %d." % (len(data_asset_name_list)))
+        else:
+            # logger.info("Profiling the first %d, alphabetically." % (max_data_assets))
+            print("Profiling the first %d, alphabetically." % (max_data_assets))
+            data_asset_name_list.sort()
+            data_asset_name_list = data_asset_name_list[:max_data_assets]
+
+        total_columns, total_expectations, total_rows, skipped_data_assets = 0, 0, 0, 0
+        total_start_time = datetime.datetime.now()
+        for name in data_asset_name_list:
+            try:
+                start_time = datetime.datetime.now()
+
+                #FIXME: There needs to be an affordance here to limit to 100 rows, or downsample, etc.
+                batch = self.get_batch(datasource_name=datasource_name, data_asset_name=name)
+
+                #Note: This logic is specific to DatasetProfilers, which profile a single batch. Multi-batch profilers will have more to unpack.
+                expectations_config, validation_result = BasicDatasetProfiler.profile(batch)
+
+                row_count = batch.shape[0]
```

Comment:
This needs to use the DataAsset API vs the Pandas API (which will not be available outside of pandas)

Reply:
(similarly, profilers will need to define what they know how to profile and have an instance check here that the thing they know how to profile is what the batch produces. Something like:
```
BasicDatasetProfiler.data_asset_class_parent = Dataset
```

And here:
```
if not isinstance(batch, Profiler.data_asset_class_parent):
  raise ProfilerError("I don't know how to process this kind of batch")
elif isinstance(batch, Dataset):
   -> use the dataset API to check for columns, rows, etc for statistics
   -> use knowledge of dataset expectations to describe things about inspection results
-> use the DataAsset API to check for anything else (that you want to report as a statistic).
```

Reply:
Updated.